js/tabRestore.js

```diff
@@ -0,0 +1,33 @@
+const MAX_TAB_HISTORY_DEPTH = 20
+var tabHistoryStack = Array()
+
+var tabHistory = {
+  push: function (tabId) {
+    var tab = tabs.get(tabId)
+    // Do not store private tabs or blank tabs
+    if (tab.private || tab.url === 'about:blank' || tab.url === '') {
+      return
+    }
+
+    if (tabHistoryStack.length < MAX_TAB_HISTORY_DEPTH) {
+      tabHistoryStack.push(tab)
+    } else {
+      tabHistoryStack.shift()
+      tabHistoryStack.push(tab)
+    }
+  },
+  restore: function () {
+    if (tabHistoryStack.length === 0) {
+      return
+    }
+
+    var newIndex = tabs.getIndex(tabs.getSelected()) + 1
+    var newTab = tabs.add(tabHistoryStack.pop(), newIndex)
+
+    addTab(newTab, {
```

Comment:
This is addressed in https://github.com/minbrowser/min/pull/360/commits/9d46b6d54428964ccdec282a30e4813d06f16b7d